IFTB.bs

```diff
@@ -0,0 +1,535 @@
+<code>OTTO</code> if it has a CFF or CFF2 table.  However, if the file is cached it should be stored
+with the <code>IFTB</code> version.  (The differing version is to help prevent an IFTB-encoded font
+from being used accidentally as a standard font, which will generally result in blank spaces where some
+glyphs should be rendered.)
+
+IFTB Table {#iftb-table}
+------------------------
+
+The IFTB table contains information needed to determine what chunks are needed for a given font subset
+description, and then to retrieve and integrate those chunks. It is similar in layout to analogous
+OpenType tables, and is accordingly described in terms of the [OpenType Specification Data
+Types](https://learn.microsoft.com/en-us/typography/opentype/spec/otff#data-types). As with OpenType
+all fields use Big Endian byte orderng.
+
+### IFTB Table Header ### {#iftb-table-header}
+
+```
+uint16     majorVersion   - set to 1
+uint16     minorVersion   - set to 0
+uint32     reserved       - not used, set to 0
+uint32     id[4]          - ID uniquely identifying this IFTB font
+uint16     flags          - not yet used
+uint32     chunkCount     - number of bins in this IFTB font
+uint32     glyphCount     - must match field in maxp table
+Offset32   CFFCharStringsOffset  - 0 if glyf-based
+Offset32   gidMapOffset
+Offset32   chunkOffsetListOffset
+Offset32   featureMapOffset
+```
+
+### IFTB Chunk Set ### {#iftb-chunk-set}
+
+The header is followed by the chunk set:
+
+```
+uint8     chunkSet[(chunkCount + 7) / 8]
+```
+
+The chunk set is a bit array indicating whether the corresponding chunk is
+present.  The bits for chunks 0 through 7 are in chunkSet[0], those for 8
+through 15 are in chunkSet[1], and so on.  Within a byte the lowest chunk index
+is represented by the 1s bit, then the 2s, then the 4s, and so on.
+
+Because chunk 0 must always be included in the initial font file, its bit is
+always set.  Other bits may also be set at the start if other chunks are <a
+href="#iftb-preloading">preloaded</a>.  This array must be updated as chunks
+are integrated into the font, and is the only part of the IFTB table that
+changes after the initial encoding into IFTB format.
+
+### IFTB Table Strings ### {#iftb-table-strings}
+
+The chunkMap is directly followed by two strings in this format:
+
+```
+uint8      stringLength
+int8       stringContent[stringLength]
+uint8      terminator    - set to 0
+```
+
+The first string is the chunkFilesURI, which is a template for forming relative the URI for a chunk
+file.  The string must contain substrings of "$1", "$2", "$3", "$4" and/or "$5", which must be replaced
+with the corresponding hexidecimal digits of the chunk index ("$1" being the ones digit, "$2" being the
+sixteens digit, and so on) to get the relative URI of the chunk. (This can then be combined with the
+URL of the initial IFTB font file to produce the absolute URL of the chunk.)
```

Comment:
What is the encoding of the strings? (eg. ascii? utf-8?)

Reply:
I think we should adopt whatever currently makes sense from a w3c perspective. I believe URIs were ASCII for a long time but I don't know if that's still true. We should also be careful about the dollar signs.

Reply:
https://datatracker.ietf.org/doc/html/rfc6570 is probably relevant here. A standardized way to have template parameters in a URL and it also talks about encoding in section 1.6.

Reply:
I looked through this. Given that IFTB would/will be a w3c spec maybe we should consult @svgeesus at a future meeting about the template format. The powers that be might prefer something more elegant than my dollar-sign expansion (although the spec allows for it). 

For the time being I've been assuming the encoding will wind up being either ASCII or whatever corresponds to an "encoded" url.

Reply:
It's should be OK to reference an IETF RFC in this spec, we reference quite a few in the patch subset specification. You can add reference like this: [[RFC6570]]. Agreed we can discuss this further at the next meeting.